tensorflow/core/kernels/dml_kernel_wrapper.cc

```diff
@@ -103,6 +103,38 @@ void DmlKernelWrapperBase::Compute(OpKernelContext* ctx) {
       return;
     }
 
+    std::vector<std::pair<int, int>> forwardIndices;
+    forwardIndices.reserve(ctx->num_outputs());
+
+    for (int i = 0; i < ctx->num_outputs(); ++i) {
+      absl::optional<int> inputIndexToForward =
+          shared_helper->GetForwardableInputIndex(ctx, output_shapes, i);
+      // If the input is considered forwardable for the op, and the shapes match
+      if (inputIndexToForward) {
+        forwardIndices.emplace_back(
+            std::make_pair(inputIndexToForward.value(), i));
+      } else {
+        // If not all outputs are forwardable, we will forward nothing and
+        // proceed with the kernel normally.
+        forwardIndices.clear();
+        break;
```

Comment:
This shouldn't be necessary anymore now that we can mix and match forwardable outputs with non-forwardable outputs.